x-pack/test/api_integration/deployment_agnostic/apis/observability/slo/helpers/transform.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { DeploymentAgnosticFtrProviderContext } from '../../../../ftr_provider_context';
+import { RoleCredentials } from '../../../../services';
+
+export type TransformHelper = ReturnType<typeof createTransformHelper>;
+
+export function createTransformHelper(
+  getService: DeploymentAgnosticFtrProviderContext['getService'],
+  roleAuthc: RoleCredentials
+) {
+  const retry = getService('retry');
+  const supertestWithoutAuth = getService('supertestWithoutAuth');
+  const samlAuth = getService('samlAuth');
+
+  return {
+    assertNotFound: async (transformId: string) => {
+      return await retry.tryWithRetries(
+        `Wait for transform ${transformId} to be deleted`,
+        async () => {
+          const response = await supertestWithoutAuth
+            .get(`/internal/transform/transforms/${transformId}`)
+            .set(roleAuthc.apiKeyHeader)
+            .set(samlAuth.getInternalRequestHeader())
+            .set('elastic-api-version', '1')
```

Comment:
For "internal" APIs authentication should be performed using the Cookie header, not with API key:

You can get cookieHeader this way:
```ts
const cookieHeader = await samlAuth.getM2MApiCookieCredentialsWithRoleScope(role);
```

or event get rid of `samlAuth` and `supertestWithoutAuth` in favour of

```
      const roleScopedSupertest = getService('roleScopedSupertest');
      let supertestWithAdminScope: SupertestWithRoleScopeType;

      supertestAdminWithCookieCredentials =
        await roleScopedSupertest.getSupertestWithRoleScope('admin', {
          withCookieHeader: true,
          withInternalHeaders: true,
        });
```

 Since transformHelper only uses internal APIs there is no need to pass `roleAuthc`, you can get Cookie header directly in method (we generate single cookie for each role per FTR config run to speed up test execution)

You can read more in FTR docs: https://github.com/elastic/kibana/blob/main/x-pack/test_serverless/README.md#api-integration-test-example

Reply:
Hope you don't mind converting it to FTR service:

```
/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import { SupertestWithRoleScopeType } from '.';
import { DeploymentAgnosticFtrProviderContext } from '../ftr_provider_context';

export function TransformApiProvider({ getService }: DeploymentAgnosticFtrProviderContext) {
  const retry = getService('retry');
  const roleScopedSupertest = getService('roleScopedSupertest');
  let supertestAdminWithCookieCredentials: SupertestWithRoleScopeType;

  const getSupertestWithCookieHeader = async () => {
    if (!supertestAdminWithCookieCredentials) {
      supertestAdminWithCookieCredentials = await roleScopedSupertest.getSupertestWithRoleScope(
        'admin',
        {
          useCookieHeader: true,
          withInternalHeaders: true,
          withCustomHeaders: { 'elastic-api-version': '1' },
        }
      );
    }
    return supertestAdminWithCookieCredentials;
  };

  return {
    assertNotFound: async (transformId: string) => {
      return await retry.tryWithRetries(
        `Wait for transform ${transformId} to be deleted`,
        async () => {
          const response = await (await getSupertestWithCookieHeader())
            .get(`/internal/transform/transforms/${transformId}`)
            .send()
            .timeout(10000)
            .expect(404);

          return response.body;
        },
        { retryCount: 5, retryDelay: 2000 }
      );
    },

    assertExist: async (transformId: string) => {
      return await retry.tryWithRetries(
        `Wait for transform ${transformId} to exist`,
        async () => {
          const response = await (await getSupertestWithCookieHeader())
            .get(`/internal/transform/transforms/${transformId}`)
            .send()
            .timeout(10000)
            .expect(200);
          return response.body;
        },
        { retryCount: 5, retryDelay: 2000 }
      );
    },
  };
}

Reply:
I don't mind changing this. My initial thought was to keep this separate from the other getServices because it could become very specific to the SLO test suite.

Reply:
I see. Then you can just use
`const cookieHeader = await samlAuth.getM2MApiCookieCredentialsWithRoleScope(role);`

to get cookie header and use it instead of API key

Reply:
addressed in https://github.com/elastic/kibana/pull/195927/commits/6b46330fb8290f854fd0753103fa184eb88ce1f0

Reply:
@kdelemme you still need to pass internal headers with       `.set(samlAuth.getInternalRequestHeader())`